### PR TITLE
feat:add agent hostNetWork options

### PR DESCRIPTION
**Pull request by chenjinxuan**

### CODE REVIEW

**Comment by objectiser** [comment]
@chenjinxuan Thank you for providing a PR,

As mentioned on #1070 this will need to wait for @jpkrohling to return from PTO next week. Its possibly this can be fixed without exposing additional options in the CR.

**Comment by chenjinxuan** [comment]
> @chenjinxuan Thank you for providing a PR,
> 
> As mentioned on #1070 this will need to wait for @jpkrohling to return from PTO next week. Its possibly this can be fixed without exposing additional options in the CR.

ok

File: pkg/apis/jaegertracing/v1/jaeger_types.go [hunk 1]
```diff
@@ -183,6 +183,9 @@ type JaegerCommonSpec struct {
 
 	// +optional
 	ServiceAccount string `json:"serviceAccount,omitempty"`
+
+	// +optional
+	HostNetWork bool `json:"hostNetWork,omitempty"`
```
Comment (jpkrohling) [inline pkg/apis/jaegertracing/v1/jaeger_types.go:188]: This should be `*bool` instead. Nil indicates that users have not set this flag explicitly in the CR.

File: .gitignore [hunk 2]
```diff
@@ -82,3 +82,4 @@ tags
 # End of https://www.gitignore.io/api/go,vim,emacs,visualstudiocode
 fmt.log
 import.log
+.idea/
```
Comment (jpkrohling) [inline .gitignore:85]: If possible, please revert this change, as it's not related to this PR.

File: pkg/deployment/agent.go [hunk 3]
```diff
@@ -120,6 +120,7 @@ func (a *Agent) Get() *appsv1.DaemonSet {
 					Annotations: commonSpec.Annotations,
 				},
 				Spec: corev1.PodSpec{
+					HostNetwork: a.jaeger.Spec.Agent.HostNetWork,
```
Comment (jpkrohling) [inline pkg/deployment/agent.go:123]: Once the value is changed to `*bool`, you'll have to nil-check the value and use a default (false).
